data.md

```diff
@@ -326,18 +248,19 @@ Byte Arrays
 
 The local memory of execution is a byte-array (instead of a word-array).
 
--   `#asUnsigned` will interpret a WordStack as a single unsigned integer (with MSB first).
+-   `#asUnsigned` will interpret a substring of a Bytes as a single unsigned integer (with MSB first).
 -   `#asAccount` will interpret a Bytes as a single account id (with MSB first).
     Differs from `Bytes2Int` only in that an empty stack represents the empty account, not account zero.
 -   `B [ N .. W ]` access the range of `B` beginning with `N` of width `W` (padding with zeros as needed).
 -   `B [ N := B' ]` sets elements starting at $N$ of $B$ to $B'$ (padding with zeros as needed).
 
 ```k
-    syntax Int ::= #asUnsigned ( WordStack ) [function]
- // ---------------------------------------------------
-    rule #asUnsigned( .WordStack )    => 0
-    rule #asUnsigned( W : .WordStack) => W
-    rule #asUnsigned( W0 : W1 : WS )  => #asUnsigned(((W0 <<Int 8) |Int W1) : WS)
+    syntax Int ::= #asUnsigned ( Int , Int , Bytes )      [function]
+                 | #asUnsigned ( Int , Int , Bytes, Int ) [function]
+ // ----------------------------------------------------------------
+    rule #asUnsigned( I, N, BS ) => #asUnsigned(I, N, BS, 0)
+    rule #asUnsigned( _, 0, _, X )  => X
+    rule #asUnsigned( I, N, BS, X ) => #asUnsigned( I +Int 1, N -Int 1, BS, (X *Int 256) +Int BS[I] ) requires N >Int 0
```

Comment:
I think you may be able to replace uses of `#asUnsigned` with uses of `Bytes2Int` from `domains.md` instead?

Reply:
This is a special function that allows you to specify a substring of a Bytes. Otherwise, you would have to do Bytes2Int(BS[ X .. Y ]) which wouldn't be a random access.

Reply:
@dwightguth would using something like `Bytes2Int(substrBytes(BS, X, Y))` cause excessive memory duplication, or is `substrBytes` efficient?

Reply:
SubstrBytes is O(N) in the length of the substring.'